backend/src/utils/db_utils.py

```
@@ -1099,99 +1097,85 @@ def list_categories_by_user_from_db(user_id: str, parent_category_id: Optional[u
+        
+        # Save updates to DynamoDB
+        logger.info(f"DIAG: Calling to_dynamodb_item...")
+        get_categories_table().put_item(Item=category.to_dynamodb_item())
+        
         logger.info(f"DB: Category {str(category_id)} updated successfully.")
-        return Category.from_dynamodb_item(response['Attributes'])
+        return category
+        
+    except ValidationError as e:
+        logger.error(f"DB: Validation error updating category {str(category_id)}: {str(e)}")
+        raise ValueError(f"Invalid update data: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)